casper/src/main/scala/coop/rchain/casper/genesis/Genesis.scala

```diff
@@ -103,6 +94,62 @@ object Genesis {
                      )
     } yield genesisBlock
 
+  private def getVaults[F[_]: Sync: Log: RaiseIOError](
+      maybeVaultPath: Option[String],
+      defaultVaultPath: Path
+  ): F[Seq[Vault]] =
+    maybeVaultPath match {
+      case Some(vaultsPathStr) =>
+        val vaultsPath = Paths.get(vaultsPathStr)
+        Monad[F].ifM(exists(vaultsPath))(
+          vaultFromFile[F](vaultsPath),
+          Sync[F].raiseError(new Exception(s"Specified vaults file $vaultsPath does not exist"))
+        )
+      case None =>
+        Monad[F].ifM(exists(defaultVaultPath))(
```

Comment:
I think by default it is `~/.rnode/genesis/wallets.txt`. @sebtomba might know more.

Reply:
My point is that we don't need two paths from which we'll read the wallets file. The config option could have default value, but if not - we go with what the user passed. Notice this would happen in the config code, not here.

Then we read the file from whatever the config value resolved to (default or not).

Then we terminate if the file is not there.

Reply:
I've created a ticket for this: https://rchain.atlassian.net/browse/RCHAIN-3590